contracts/v3/mock/MockConvexVault.sol

```diff
@@ -0,0 +1,167 @@
+
+    address public owner;
+    address public immutable staker;
+    address public immutable minter;
+
+    struct PoolInfo {
+        address lptoken;
+        address token;
+        address gauge;
+        address crvRewards;
+        address stash;
+        bool shutdown;
+    }
+
+    //index(pid) -> pool
+    PoolInfo[] public poolInfo;
+
+    event Deposited(address indexed user, uint256 indexed poolid, uint256 amount);
+    event Withdrawn(address indexed user, uint256 indexed poolid, uint256 amount);
+
+    constructor(
+        address _staker,
+        address _minter,
+        address _crv,
+        address _cvx
+    ) public {
+        staker = _staker;
+        owner = msg.sender;
+        minter = _minter;
+        crv = _crv;
+        cvx = _cvx;
+    }
+
+    function poolLength() external view returns (uint256) {
+        return poolInfo.length;
+    }
+
+    //create a new pool
+    function addPool(
+        address _lptoken,
+        address _gauge,
+        uint256 _stashVersion
+    ) external returns (bool) {
+        //the next pool's pid
+        uint256 pid = poolInfo.length;
+
+        //create a tokenized deposit
+        MockERC20 token = new MockERC20(
+            string(abi.encodePacked(ERC20(_lptoken).name(), ' Convex Deposit')),
+            string(abi.encodePacked('cvx', ERC20(_lptoken).symbol())),
+            18
+        );
+
+        //create a reward contract for crv rewards
+        MockBaseRewardPool newRewardPool = new MockBaseRewardPool(
+            pid,
+            address(token),
+            crv,
+            address(this),
+            address(this)
+        );
+
+        // give some fake generated rewards to reward pool so user can claim
+        IERC20(cvx).safeTransferFrom(address(this), address(newRewardPool), 10000);
+
+        //add the new pool
+        poolInfo.push(
+            PoolInfo({
+                lptoken: _lptoken,
+                token: address(token),
+                gauge: _gauge,
+                crvRewards: address(newRewardPool),
+                stash: address(0),
+                shutdown: false
+            })
+        );
+        return true;
+    }
+
+    //deposit lp tokens and stake
+    function deposit(
+        uint256 _pid,
+        uint256 _amount,
+        bool _stake
+    ) public returns (bool) {
+        PoolInfo storage pool = poolInfo[_pid];
+        //send to proxy to stake
+        address lptoken = pool.lptoken;
+        IERC20(lptoken).safeTransferFrom(msg.sender, address(this), _amount);
+
+        address token = pool.token;
+        /*if (_stake) {
+            //mint here and send to rewards on user behalf
+            MockERC20(token).mint(address(this), _amount);
+            address rewardContract = pool.crvRewards;
+            IERC20(token).safeApprove(rewardContract, 0);
+            IERC20(token).safeApprove(rewardContract, _amount);
+            MockBaseRewardPool(rewardContract).stakeFor(msg.sender, _amount);
+        } else {*/
```

Comment:
If this isn't needed, remove commented-out code.

Reply:
I wasn't sure whether we needed to mock the `_stake` path or not or if it would suffice just going the normal route and using the fixed reward amount from `MockConvexBaseRewardPool.getReward` that I implemented.